src/sentry/auth/helper.py

```diff
@@ -440,7 +440,7 @@ def handle_unknown_identity(
                 try:
                     self._login(acting_user)
                 except self._NotCompletedSecurityChecks:
-                    if acting_user.has_usable_password():
+                    if acting_user.has_usable_password() or is_account_verified:
```

Comment:
Can you write a comment + add a test for this? 

This is one of those subtle edge-cases and I'd prefer if it's behaviour is documented.